sources/packages/web/src/components/common/NoticeOfAssessmentFormView.vue

```diff
@@ -150,62 +154,67 @@ export default defineComponent({
-      noaDisbursementSchedule.disbursement2MSFAACancelledDateFormatted =
-        dateOnlyLongString(
+        noaDisbursementSchedule.disbursement1MSFAAStatus = firstMSFAAStatus;
+        noaDisbursementSchedule.disbursement1MSFAACancelledDateFormatted =
+          dateOnlyLongString(
+            noaDisbursementSchedule.disbursement1MSFAACancelledDate,
+          );
+        noaDisbursementSchedule.disbursement1MSFAAStatusClass =
+          getMSFAAStatusClass(firstMSFAAStatus);
+        // Second disbursement.
+        const secondMSFAAStatus = getMSFAAStatus(
+          noaDisbursementSchedule.disbursement2MSFAADateSigned,
           noaDisbursementSchedule.disbursement2MSFAACancelledDate,
         );
-      noaDisbursementSchedule.disbursement2MSFAAStatusClass =
-        getMSFAAStatusClass(secondMSFAAStatus);
-      // Checks if the component allows the MSFAA reissue and if there is
-      // one pending disbursement with a cancelled MSFAA.
-      const canReissueMSFAA =
-        props.canReissueMSFAA &&
-        ((noaDisbursementSchedule.disbursement1Status ===
-          DisbursementScheduleStatus.Pending &&
-          !!noaDisbursementSchedule.disbursement1MSFAACancelledDate) ||
-          (noaDisbursementSchedule.disbursement2Status ===
+        noaDisbursementSchedule.disbursement2MSFAAStatus = secondMSFAAStatus;
+        noaDisbursementSchedule.disbursement2MSFAACancelledDateFormatted =
+          dateOnlyLongString(
+            noaDisbursementSchedule.disbursement2MSFAACancelledDate,
+          );
+        noaDisbursementSchedule.disbursement2MSFAAStatusClass =
+          getMSFAAStatusClass(secondMSFAAStatus);
+        // Checks if the component allows the MSFAA reissue and if there is
+        // one pending disbursement with a cancelled MSFAA.
+        const canReissueMSFAA =
+          props.canReissueMSFAA &&
+          ((noaDisbursementSchedule.disbursement1Status ===
             DisbursementScheduleStatus.Pending &&
-            !!noaDisbursementSchedule.disbursement2MSFAACancelledDate));
-      initialData.value = {
-        ...assessment,
-        canReissueMSFAA,
-      };
-      emit(
-        "assessmentDataLoaded",
-        initialData.value.applicationStatus,
-        initialData.value.noaApprovalStatus,
-        initialData.value.applicationCurrentAssessmentId,
-      );
+            !!noaDisbursementSchedule.disbursement1MSFAACancelledDate) ||
+            (noaDisbursementSchedule.disbursement2Status ===
+              DisbursementScheduleStatus.Pending &&
+              !!noaDisbursementSchedule.disbursement2MSFAACancelledDate));
+        initialData.value = {
+          ...assessment,
+          canReissueMSFAA,
+        };
+        emit(
+          "assessmentDataLoaded",
+          initialData.value.applicationStatus,
+          initialData.value.noaApprovalStatus,
+          initialData.value.applicationCurrentAssessmentId,
+        );
+      }
     };
 
     watchEffect(loadNOA);
```

Comment:
As a suggestion, to keep the code cleaner I would keep the `loadNOA` as it is and use the watch effect to apply the logic. 
![image](https://github.com/user-attachments/assets/75b015db-4ebb-4823-952b-d6ffe07c6b18)